### ISSUE TITLE
Bug in pg_loss_with_log_probs

### ISSUE THREAD

**Issue by gpengzhi** [asyml/texar#159]
Should the code in https://github.com/asyml/texar/blob/master/texar/losses/pg_losses.py#L241
be `losses = tf.reduce_mean(losses, axis=list(range(1, rank)))`?
